flake.nix

```diff
@@ -91,8 +91,11 @@
             makeWrapper
             pkg-config
 
+            # for libz-ng-sys (zlib-ng)
+            cmake
```

Comment:
Ugh, I'm not a huge fan of this. It looks like by default it will actually cause a full build of the library in the background, based on my reading of the build file? We already have `zlib-ng` packaged fully in Nixpkgs, so I'd rather avoid that if possible. I'll have to take a closer look later today.

What's the performance impact from this? I'm figure it's reasonably significant (zlib-ng really is much faster) but I'm curious.

Reply:
I don't know nix, so it'll be nice if you can figure out what we should do here.

> What's the performance impact from this?

See the commit message. It has some measurable difference in file/annotate query.

Reply:
It seems like `libz-ng-sys`’s build script is tragically unwilling to use a system package, even though `libz-sys` does the right thing and checks `pkg-config`.

@Byron Would you accept a patch to `libz-ng-sys` to move the `zlib-ng` logic in the build script so that it can use system packages for that library as well where available?

Reply:
Yes, if we can get it to just use the system libraries like we do with `zstd` and `libssh2` (`export ZSTD_SYS_USE_PKG_CONFIG=1` for example), then that should be sufficient to just drop the `cmake` input and use `zlib-ng` instead.

Reply:
`libz-sys` doesn’t support an environment variable in general, unfortunately (or we’d export it from our Rust setup hook). But it does look for system `zlib` first, and `libz-ng-sys` is maintained in the same repository with the same build script, so I am hoping it will be a simple matter of sending a PR for the `build.rs` and bumping to a new Gitoxide release. Then we can use the Nixpkgs `zlib-ng` package and everyone is happy.

Reply:
> @Byron Would you accept a patch to `libz-ng-sys` to move the `zlib-ng` logic in the build script so that it can use system packages for that library as well where available?

It seems like this should be done, even though I don't know if there is any reason this wasn't done.
Ideally, the current behaviour remains, and a flag of sorts is added to enable looking for the system package. 

From what I know, `libz-ng-sys` uses its own build-script as well.

Reply:
`build.rs` calls out to some `libz-ng-sys`‐specific code, but the actual file itself is shared, and that’s the file that contains the `pkg-config` logic, so I figured it would just need some moving around and conditional names.

Would you be amenable to an environment variable that forces the use of `pkg-config`, for both `libz-sys` and `libz-ng-sys`? That would solve the current problem and be convenient in general for Nixpkgs, as most other packages that vendor libraries support it, and the current default behaviour of proactively searching `pkg-config` for `libz-sys` but not for `libz-ng-sys` could be retained for backwards compatibility.

Reply:
Yes, environment variables seem to be the most common way to do that and keep it backwards compatible with it. Thank you.

Reply:
Thanks! I’ll try to send a PR off today.

Reply:
Is this flake.nix issue a blocker? Can we address it later? or add feature flag to disable zlib-ng on nix?

Reply:
Let's just merge it for now; I'd rather not have a true "parity" distinction between Nix users and others. We can look at fixing `zlib-ng` in a proper way this release cycle.